### PR TITLE
Sort Trainer arguments based on importance

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

I propose to organize the Trainer's arguments based on importance/usage frequency. This is just my take on it, please make suggestions. Alternatively, they could also be grouped by function. 

Before, the arguments were arranged randomly. 

cc @borda @justusschock @awaelchli

### CODE REVIEW

File: src/lightning/pytorch/trainer/trainer.py [hunk 1]
```diff
@@ -89,19 +89,15 @@ class Trainer:
     @_defaults_from_env_vars
     def __init__(
         self,
+        *,
```
Comment (awaelchli) [inline src/lightning/pytorch/trainer/trainer.py:92]: Forcing keyword-only usage, because not doing so can be very error-prone.

**Review by carmocca — APPROVED** [review]

File: tests/tests_pytorch/callbacks/progress/test_rich_progress_bar.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ def test_rich_progress_bar_custom_theme(tmpdir):
         theme = RichProgressBarTheme()
 
         progress_bar = RichProgressBar(theme=theme)
-        progress_bar.on_train_start(Trainer(tmpdir), BoringModel())
+        progress_bar.on_train_start(Trainer(default_root_dir=tmpdir), BoringModel())
```
Comment (awaelchli) [inline tests/tests_pytorch/callbacks/progress/test_rich_progress_bar.py:117]: we were passing `Trainer(logger=tmpdir)` 🤣

Comment (carmocca) [reply tests/tests_pytorch/callbacks/progress/test_rich_progress_bar.py:117]: It clearly did not run then. I suggest removing the tmpdir fixture instead for that 5ms time save

**Review by justusschock — APPROVED** [review]

File: tests/tests_pytorch/callbacks/progress/test_rich_progress_bar.py [hunk 3]
```diff
@@ -102,7 +102,7 @@ def test_rich_progress_bar_import_error(monkeypatch):
```
Comment (akihironitta) [inline tests/tests_pytorch/callbacks/progress/test_rich_progress_bar.py:102]: the commit msg 🤣

**Comment by Borda** [comment]
this is a breaking change so a user can't pass just an argument

> Before, the arguments were arranged randomly.

not randomly but with respect to minimizing breaking changes.... :)

**Pull request merged**